### PR TITLE
Fix/set log url anytime

### PR DESCRIPTION
**Pull request by clpolet**
Allow updating the log URL at runtime. This is done in the base HTTP sender, will need to do some small work to port this to base/updates branch.

### CODE REVIEW

File: MobileCenter/MobileCenter/Internals/Sender/MSHttpSender.h [hunk 1]
```diff
@@ -6,9 +6,14 @@ NS_ASSUME_NONNULL_BEGIN
...
 /**
+ * Base URL (schema + authority + port only) used to communicate with the server.
+ */
+@property(nonatomic) NSString *baseURL;
+
+/**
  *	Send Url.
  */
-@property(nonatomic, strong, readonly) NSURL *sendURL;
+@property(nonatomic, strong) NSURL *sendURL;
```
Comment (TroubleMakerBen) [inline MobileCenter/MobileCenter/Internals/Sender/MSHttpSender.h]: we don't need to specify `strong` here. Should we add "Do not specify default attributes for properties" as a coding rule?

Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Sender/MSHttpSender.h]: Why not. I know there are two schools about this, people that wants everything specified to be explicit but some other prefer the concise way for aesthetic.

Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Sender/MSHttpSender.h]: Ok so we'll skip the defaults for now on. But for consistency we'll do that on another PR for the whole code.

Comment (TroubleMakerBen) [inline MobileCenter/MobileCenter/Internals/Sender/MSHttpSender.h]: Yep.

File: MobileCenter/MobileCenter/Internals/MSMobileCenterInternal.h [hunk 2]
```diff
@@ -1,8 +1,8 @@
-#import <Foundation/Foundation.h>
```
Comment (jaelim-ms) [inline MobileCenter/MobileCenter/Internals/MSMobileCenterInternal.h]: OK. So what is a rule for `#import <header>` and `#import "header"`? I remember we do `<>` and then `""`.

File: MobileCenter/MobileCenter/Internals/Sender/MSHttpSender.m [hunk 3]
```diff
@@ -62,7 +68,12 @@ - (id)initWithBaseUrl:(NSString *)baseUrl
          queryStrings:(NSDictionary *)queryStrings
          reachability:(MS_Reachability *)reachability
        retryIntervals:(NSArray *)retryIntervals {
-  return [self initWithBaseUrl:baseUrl apiPath:@"" headers:headers queryStrings:queryStrings reachability:reachability retryIntervals:retryIntervals];
+  return [self initWithBaseUrl:baseUrl
```
Comment (jaelim-ms) [inline MobileCenter/MobileCenter/Internals/Sender/MSHttpSender.m]: Not related to your change but we can remove this method.

File: MobileCenter/MobileCenter/Internals/Sender/MSSender.h [hunk 4]
```diff
@@ -11,7 +11,17 @@ NS_ASSUME_NONNULL_BEGIN
 @protocol MSSender <NSObject, MSSenderCallDelegate, MSEnable>
 
 /**
- *  Reachability library.
+ * Base URL (schema + authority + port only) used to communicate with the server.
```
Comment (jaelim-ms) [inline MobileCenter/MobileCenter/Internals/Sender/MSSender.h]: This is a generic sender. I don't think we need to place those properties in here. Need to go to http sender, what do you think?

Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Sender/MSSender.h]: I moved apiPath which is only used by senders.

File: MobileCenter/MobileCenter/Internals/Sender/MSHttpSender.m (hunk 3 above)
Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Sender/MSHttpSender.m]: done

File: MobileCenter/MobileCenter/Internals/MSMobileCenterInternal.h (hunk 2 above)
Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/MSMobileCenterInternal.h]: done

File: MobileCenter/MobileCenter/Internals/Sender/MSSender.h (hunk 4 above)
Comment (jaelim-ms) [inline MobileCenter/MobileCenter/Internals/Sender/MSSender.h]: As per offline conversation, you need to make one more change for baseURL.
